pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -1265,12 +1273,18 @@ func hashForService(push *model.PushContext,
 func hashForVirtualService(push *model.PushContext,
 	node *model.Proxy,
 	virtualService config.Config,
+	serviceRegistry map[host.Name]*model.Service,
 ) (DestinationHashMap, []*model.ConsolidatedDestRule) {
 	hashByDestination := DestinationHashMap{}
 	destinationRules := make([]*model.ConsolidatedDestRule, 0)
 	for _, httpRoute := range virtualService.Spec.(*networking.VirtualService).Http {
 		for _, destination := range httpRoute.Route {
-			hash, dr := hashForHTTPDestination(push, node, destination)
+			hostname := host.Name(destination.GetDestination().GetHost())
+			var svc *model.Service
+			if serviceRegistry != nil {
+				svc = serviceRegistry[hostname]
```

Comment:
@ramaraochavali thanks for taking a look at this.  This is the part that I was hesitant about since I wasn't sure of the performance impacts for the additional service lookup

Reply:
This is is not a perf impacting look up. We already have a map. It is just a map look up